### ISSUE TITLE
"invalid character in header" Exception with some websites

### ISSUE THREAD

**Issue by Ins1ghtLabs** [gwik/geventhttpclient#63]
for example http://papileon.at/
Any web browser can open it with no problem.

But...

ua = UserAgent()

> > > r=ua.urlopen("http://papileon.at/")
> > > Traceback (most recent call last):
> > >   File "<stdin>", line 1, in <module>
> > >   File "/usr/lib64/python2.7/site-packages/geventhttpclient/useragent.py", line 332, in urlopen
> > >     e = self._handle_error(e, url=req.url)
> > >   File "/usr/lib64/python2.7/site-packages/geventhttpclient/useragent.py", line 327, in urlopen
> > >     resp = self._urlopen(req)
> > >   File "/usr/lib64/python2.7/site-packages/geventhttpclient/useragent.py", line 274, in _urlopen
> > >     body=request.payload, headers=request.headers)
> > >   File "/usr/lib64/python2.7/site-packages/geventhttpclient/client.py", line 167, in request
> > >     block_size=self.block_size, method=method.upper(), headers_type=self.headers_type)
> > >   File "/usr/lib64/python2.7/site-packages/geventhttpclient/response.py", line 277, in __init__
> > >     super(HTTPSocketPoolResponse, self).**init**(sock, **kw)
> > >   File "/usr/lib64/python2.7/site-packages/geventhttpclient/response.py", line 149, in **init**
> > >     self._read_headers()
> > >   File "/usr/lib64/python2.7/site-packages/geventhttpclient/response.py", line 170, in _read_headers
> > >     self.feed(data)
> > > _parser.HTTPParseError: ('invalid character in header', 21)

**Comment by ml31415** [comment]
This is the header string, that your server is sending:

```
HTTP/1.1 200 OK
Server: nginx/1.2.5
Date: Thu, 26 Mar 2015 22:08:48 GMT
Content-Type: text/html; charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
Vary: Accept-Encoding
X-Powered-By: PHP/5.3.18-1~dotdeb.0
Set-Cookie: fe_typo_user=5bc4be835ac7364f9f48811af633d845; path=/
X-UA-Compatible "IE=edge,chrome=1": 
Content-Encoding: gzip
```

The parser chokes on the `X-UA-Compatible` line with invalid format. So, so the parser is right to complain. The question is, if it should fail, or just ignore the line.

**Comment by yanivhdd** [comment]
Should be a mode to ignore bad lines in header with default false

**Comment by ml31415** [comment]
Pull requests welcome.

**Comment by ml31415** [comment]
Do you mind adding a test case to proof the fix in the PR?

**Comment by cyberw** [comment]
Maybe fixed in #81 ? Closing.
